uji.py

```diff
@@ -1092,6 +1092,8 @@ def toggle(self):
         else:
             self.unmark()
 
+        self.next()
```

Comment:
Shouldn't this be called also after running a program?

It just feels wrong to have a different behavior, but OTOH, "running" doesn't mean that we are done with the test, just that we executed it.

And now that I come to it, should this be only valid for skip/pass/fail?

Reply:
yep, let's do that. Was missing because this patchset had to be rebased on top of the run patchset, so it was missing from there